xdsl/dialects/llvm.py

```diff
@@ -144,6 +145,49 @@ def from_size_and_type(size: int | IntAttr, type: Attribute):
         return LLVMArrayType([size, type])
 
 
+@irdl_attr_definition
+class LinkageAttr(ParametrizedAttribute):
+    name = "llvm.linkage"
+
+    linkage: ParameterDef[StringAttr]
+
+    def print_parameters(self, printer: Printer) -> None:
+        printer.print_string("<")
+        printer.print_attribute(self.linkage)
+        printer.print_string(">")
+
+    @staticmethod
+    def parse_parameters(parser: BaseParser) -> list[Attribute]:
+        parser.parse_characters('<', "llvm.linkage parameter expected")
+        # The linkage string is output from xDSL as a string (and accepted by MLIR as such)
+        # however it is always output from MLIR without quotes. Therefore need to determine
+        # whether this is a string or not and slightly change how we parse based upon that
+        linkage_str = parser.try_parse_string_literal()
+        if linkage_str is not None:
+            linkage_str = linkage_str.string_contents
+        else:
+            linkage_str = parser.tokenizer.next_token().text
+        linkage = StringAttr(linkage_str)
+        parser.parse_characters('>', "End of llvm.linkage parameter expected!")
+        return [linkage]
+
+    @staticmethod
+    def from_linkage(linkage: str | StringAttr):
```

Comment:
nit: I'm not sure we need this constructor, it seems clearer to write `LinkageAttr(my_str)` or `LinkageAttr(my_attr)`